src/pid/environ.rs

```diff
@@ -8,59 +8,148 @@ use std::path::Path;
 use std::iter::Iterator;
 
 use libc::pid_t;
-use nom::IResult;
+use nom::{self, IResult};
 
+/// An environment holder.
 #[derive(Debug, Clone)]
 pub struct Environ {
     data: Vec<u8>,
 }
 
-impl Environ {
-    fn parse(&self) -> Option<Result<(Vec<u8>, (OsString, OsString))>> {
-        if self.data.is_empty() {
-            // Don't attempt to parse an empty file.
-            return None;
-        }
-        // Map the result of `take_until_and_consume` to OsString.
-        named_args!(
-            until<'a>(sep: &'a str)<OsString>,
-            map!(take_until_and_consume!(sep), from_bytes)
-        );
-        // Parse 'key=value" pair.
-        named!(
-            pair<&[u8], (OsString, OsString)>,
-            tuple!(apply!(until, "="), apply!(until, "\0"))
-        );
-        match pair(&self.data) {
-            IResult::Done(leftover, parsed) => Some(Ok((leftover.to_vec(), parsed))),
+/// An iterator over environment variables.
+pub struct EnvironIter<'a> {
```

Comment:
I'm in doubt about naming and whether the structure should be exported to a top-level (`pid::EnvironIter`) or not (I think not).